### PR TITLE
Final changes to introduce mem_desc to be hold in Tensor

### PR DESCRIPTION
**Pull request by jczaja**
### PR types
Others 

### PR changes
Others 

### Describe
This PR removed set_format() and format() from Tensor and its functionality is replaced by set_mem_desc() and mem_desc() across codebase.

### CODE REVIEW

**Review by jakpiase — DISMISSED** [review]
LGTM

**Comment by onecatcn** [comment]
@jczaja thank you very much for the contribution. are you still working on this PR? Should I wait for you to fix those CI failures before I contact colleagues for further checks?

**Comment by jczaja** [comment]
@onecatcn  Hi, This PR depends on https://github.com/PaddlePaddle/Paddle/pull/47617 . So after https://github.com/PaddlePaddle/Paddle/pull/47617 is merged I can resume work on this one.

**Comment by onecatcn** [comment]
> 

I see. thank you for the explaination. https://github.com/PaddlePaddle/Paddle/pull/47617 is merged now.

File: paddle/fluid/operators/mkldnn/reshape_mkldnn_op.cc [hunk 1]
```diff
@@ -83,23 +82,22 @@ class ReshapeMKLDNNKernel : public framework::OpKernel<T> {
...
-        x->format(), platform::to_void_cast(x->data<T>()));
+        x->mem_desc(), platform::to_void_cast(x->data<T>()));
     out->Resize(x_dims);  // to match x numel, format is changed later
     // reorder is done into a plain tag to allow usage with blocked formats
     auto reorder_dst_memory_p = reorder_handler.AcquireDstMemory(
         out, getPlainFormatTag(x), ctx.GetPlace());
-    auto reorder_p = reorder_handler.AcquireReorder(reorder_src_memory_p,
-                                                    reorder_dst_memory_p);
+    auto reorder_p = reorder_handler.AcquireReorder(reorder_dst_memory_p,
+                                                    reorder_src_memory_p);
```
Comment (wozna) [inline paddle/fluid/operators/mkldnn/reshape_mkldnn_op.cc:91]: Did it work properly before?

Comment (jczaja) [reply paddle/fluid/operators/mkldnn/reshape_mkldnn_op.cc:91]: that was one of problems found

Comment (jczaja) [reply paddle/fluid/operators/mkldnn/reshape_mkldnn_op.cc:91]: @jakpiase  please comment on that one as I think you found it

Comment (jakpiase) [reply paddle/fluid/operators/mkldnn/reshape_mkldnn_op.cc:91]: No, previously it was bugged. Switching the order of memories will result in proper execution

**Review by jakpiase — APPROVED** [review]
LGTM

**Comment by jczaja** [comment]
@wozna Please continue your review

**Review by wozna — APPROVED** [review]
LGTM

**Pull request merged**